### PR TITLE
Rename kernel-list-cache directory to follow underscore pattern

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
rename kernel-list-cache directory to follow underscore pattern

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

**Review by Tulsishah — DISMISSED** [review]

**Review by raj-prince — APPROVED** [review]

File: tools/integration_tests/run_e2e_tests.sh [hunk 1]
```diff
@@ -63,7 +63,8 @@ TEST_DIR_PARALLEL=(
   "interrupt"
   "operations"
   "log_content"
-  "kernel-list-cache"
+  "kernel_list_cache"
+  "concurrent_operations"
```
Comment (kislaykishore) [inline tools/integration_tests/run_e2e_tests.sh:67]: nit: Is "concurrent_operations" expected in the diff? Should it be in a different PR?

Comment (ashmeenkaur) [reply tools/integration_tests/run_e2e_tests.sh:67]: Yeah, should have done this in a separate PR. I'll take care next time.

**Review by vadlakondaswetha — APPROVED** [review]
